sdb/commands/zfs/histograms.py

```diff
@@ -0,0 +1,149 @@
+#
+# Copyright 2020 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+import argparse
+from typing import Iterable
+
+import drgn
+import sdb
+from sdb.commands.internal import fmt
+
+
+class ZFSHistogram(sdb.Command):
+    """
+    Print ZFS Histogram and print its median segment size.
+
+    NOTE
+        The median is just an approximation as we can't tell the
+        exact size of each bucket within a histogram bucket.
+
+    EXAMPLES
+        Dump the histogram of the normal metaslab class of the rpool:
+
+            sdb> spa rpool | member spa_normal_class.mc_histogram | zhist
+            seg-size   count
```

Comment:
Is this command intended to be generic enough to be used with any array of integers? If not, what is the dependency on ZFS? If so, does the `seg-size` name make sense for other (e.g. non metaslab) contexts? Along the same lines, does the `zhist` name make sense, if this can be used with inputs unrelated to ZFS?

Reply:
I do believe that there is more generic functionality that can come out of this but given that we are printing the indexes as nicenum'ed values shifted by an offset I'd say the command itself is ZFS specific.

We could create a generic function that optionally hides unused buckets on both ends of the histogram as we do and allows you to give it callback to manipulate the index in some way before printing it.

That said abstracting all of this seems a bit too much to me as this is a few lines of code, and most of its complexity would be in preprocessing and making sure the spacing looks nice (e.g. like `sdb/commands/internal/table.py`). Since I haven't come across any other use case where something like that would be useful I am leaning towards leaving this as is but if people think it would be useful I can give this a shot.